hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/EnsureAliasesStep.java

```diff
@@ -179,4 +181,38 @@ public static boolean isAlias(@NonNull AccountID accountID) {
         requireNonNull(accountID);
         return accountID.hasAlias() && (!accountID.hasAccountNum() || accountID.accountNum() == 0L);
     }
+
+    /**
+     * Returns the number of auto-associations implied by the given alias in the given
+     * {@link com.hedera.hapi.node.token.CryptoTransferTransactionBody}.
+     *
+     * @param aliasedReceiverId the aliased receiver ID
+     * @param op the crypto transfer transaction body
+     * @return the number of auto-associations implied by the given alias
+     */
+    public static int impliedAutoAssociationsForAlias(
+            @NonNull AccountID aliasedReceiverId, @NonNull CryptoTransferTransactionBody op) {
+        requireNonNull(op);
+        requireNonNull(aliasedReceiverId);
+        int ans = 0;
+        for (final var tokenTransferList : op.tokenTransfersOrElse(emptyList())) {
+            boolean impliedAutoAssociationHere = false;
+            for (final var unitAdjust : tokenTransferList.transfersOrElse(emptyList())) {
+                if (unitAdjust.amount() > 0 && unitAdjust.accountIDOrThrow().equals(aliasedReceiverId)) {
+                    impliedAutoAssociationHere = true;
+                    break;
+                }
+            }
+            for (final var nftTransfer : tokenTransferList.nftTransfersOrElse(emptyList())) {
```

Comment:
This for loop could be skipped if `impliedAutoAssociationHere` is true here.

Reply:
If we entered the first loop this must be a fungible token and hence will not have NFT transfers set. Added comment to clarify.